impress-org/givewp
# Fun: honeypot field

Resolves [GIVE-1160]

## Description



This adds a simple honey pot field to GiveWP v3 forms.


This can be enabled with a filter:

```php
add_filter('givewp_donation_forms_honeypot_enabled', '__return_true')
```

Per form:

```php
add_filter('givewp_donation_forms_honeypot_enabled', function($enabled, $formId){
    if ($formId === 1) {
        $enabled = true;
    }

    return $enabled;
}, 10, 2);
```

The next step would be to create a UI setting in GiveWP to enable either per-form and/or globally.

## Affects



- v3 donation forms

## Visuals



<img width="1278" alt="Screenshot 2024-08-30 at 9 17 32 AM" src="https://github.com/user-attachments/assets/2a74c629-ecff-448f-93c7-e74fa605e142">

<img width="649" alt="Screenshot 2024-08-30 at 9 18 00 AM" src="https://github.com/user-attachments/assets/6725e07b-a2b2-4a64-9c3a-3bcf2d4fb327">


## Testing Instructions



- Make a donation form request
- Add a value to the "donationBirthday" field
- You should get an error

## Pre-review Checklist






[GIVE-1160]: https://stellarwp.atlassian.net/browse/GIVE-1160?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
